src/utils/ssmCommand.ts

```diff
@@ -43,6 +37,29 @@ function sendCommand(command, instanceId): Promise<SSM.Types.SendCommandResult>
     }).promise();
 }
 
+function delay(value) {
+    return new Promise((resolve) => {
+       setTimeout(() => resolve(value), 10000)
+    });
+}
+
+const terminalState = new Set(['Success', 'TimedOut', 'Failed', 'Cancelled']);
+
+function awaitCommandResult(commandId: string, instanceId: string, remainingRetries: number): Promise<SSM.Types.GetCommandInvocationResult> {
+    return getCommandResult(commandId, instanceId)
+        .then((result: SSM.Types.GetCommandInvocationResult) => {
+            if (terminalState.has(result.Status) || remainingRetries <= 0) {
+                console.log(`Final state: ${result.Status}; remaining retries: ${remainingRetries}`);
+                return result;
+            } else {
+                console.log(`Retrying getCommandResult. Current state: ${result.Status}; remaining retries: ${remainingRetries}`);
+                return new Promise(res => setTimeout(res, 10000)).then(() => {
```

Comment:
I think we might be able to reuse the `delay` function here?

Reply:
I thought so too! But then @rtyley very helpfully explained why it doesn't work here - the `delay` function takes a value which is immediately evaluated, whereas creating this new Promise enables us to pass in a function which will only be invoked after the 10 second timeout has elapsed.